pyvista/core/grid.py

```diff
@@ -255,7 +255,7 @@ def _from_arrays(
         self._update_dimensions()
 
     @property
-    def meshgrid(self) -> list[NumpyArray[float]]:
+    def meshgrid(self) -> tuple[NumpyArray[float], ...]:
```

Comment:
Similar comment about typing based on numpy versions here.  It is a little more impactful to users however, as it is user facing and can be wrong depending on numpy version.

I'm not sure that we want to override the type to always be `tuple` however, as suggested in https://github.com/pyvista/pyvista/pull/6268#discussion_r1644906529, since this function is declared to be a simple wrapper of `np.meshgrid`.

Reply:
This is now typed with _both_ types so it is correct for both versions of numpy.

Reply:
Ah, I missed this thread, sorry. Independent discovery of always converting to `tuple` :P

Reply:
Isn't this always a 3-length tuple?
```suggestion
    def meshgrid(self) -> tuple[NumpyArray[float], NumpyArray[float], NumpyArray[float]]:
```
(modulo autoformatting)